ds0_updater.py

```diff
@@ -5,13 +5,36 @@
 import logger
 import conf
 
+# keep it super simple for now because we don't really implement --allowed-keys yet.
+def verify_latest_commit():
+    allowed_keys = ds0_utils.allowed_keys()
+
+    # ensure the latest commit is signed and verified
+    while True:
+        for key in allowed_keys:
+            logger.exe('gpg --import ' + key['file'], expectError=True)
+        output = logger.exe('git log --pretty="format:%G?" --show-signature HEAD^..HEAD')
+
+        if "Can't check signature" in output[0]:
+            logger.info('gpg keys cleared on startup. Trying again...')
+            continue
+
+        verified = False
+        for key in allowed_keys:
+            if key['rsa_check'] in output[0] and key['signature_check'] in output[0]:
+                verified = True
+        if not verified:
+            logger.error('Scripts using a non-signed commit. Please ensure commit is valid.')
+            logger.error('    If it was a missed signature, feel free to open a ticket at https://github.com/riptano/ComboAMI.')
+        break
+
 # Update the AMI codebase if it's its first boot
 if not conf.get_config("AMI", "CompletedFirstBoot"):
     # check if a specific commit was requested
     force_commit = ds0_utils.required_commit()
 
     # update the repo
-    logger.exe('git pull')
+    logger.exe('git pull ' + ds0_utils.repository())
```

Comment:
> The only caveat here is that branch-names need to be prefixed with the remote name, which can be documented and hardcoded to origin. Or perhaps leave origin in place and hardcode name for the remote of the custom-repo to somthing like 'custom-repo'.
> 
> I'm not in love with the fiddliness of the hard-coded remote name bubbling up in the user-interface, but this can reference any branch-name or commit-id, has only a single workspace transformation, and makes --forcecommit/second-half-of-repository to be a synonym while maintaining backward compat. All of which feels pretty clean.

Doesn't the assumption that the remote name is origin relieve us from bubbling it up to the user interface? After all, wherever the user would put "origin" we can have them omit it and add it implicitly.

Otherwise it seems to be the right thing to do - of course we'll run tests to make sure it all works :) - but I think we're on the right track here.